src/arrayMethodSort.js

```diff
@@ -4,8 +4,22 @@
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  const defaultCompare = function defaultCompare(a, b) {
+    return a.toString() > b.toString();
+  };
+
+  [].__proto__.sort2 = function (compareFunction = defaultCompare) {
+    const compare = compareFunction;
+
+    for (let i = 0; i < this.length - 1; i++) {
+      for (let j = 0; j < this.length - 1 - i; j++) {
+        if (compare(this[j], this[j + 1]) > 0) {
```

Comment:
The comparison function is supposed to return a number, but your implementation might still work with a boolean. However, it's not strictly following the ECMAScript standard for sort comparison functions. It's important to ensure that the comparison function returns a number to adhere to the standard.